File: DataStructsAndAlgos/implementations/javascript/src/Queue/Queue.test.js
import { ASSERT_DIRECT_COMPARE as ASSERT, logError } from "../util/ASSERT";
import { Queue } from "./Queue";

(function main() {
  console.log("##################### QUEUE ########################");
  try {
    //Write Test Code Here
    let myQueue = new Queue();
    myQueue.push(3);
    myQueue.push(2);
    myQueue.push(1);
    ASSERT(myQueue.peek(), 3);
    ASSERT(myQueue.pop(), 3);
    ASSERT(myQueue.pop(), 2);
    ASSERT(myQueue.pop(), 1);
    ASSERT(myQueue.length(), 0);
    console.log(`# # # Test 1 PASSED`);
  } catch (error) {
    logError(error, "# # # Test 1 FAILED!");
  }
  try {
    //Write Test Code Here
    let myQueue = new Queue();
    myQueue.push(1);
    myQueue.push(2);
    myQueue.push(3);
    ASSERT(myQueue.peek(), 1);
    ASSERT(myQueue.pop(), 1);
    ASSERT(myQueue.pop(), 2);
    ASSERT(myQueue.pop(), 3);
    ASSERT(myQueue.length(), 0);
    console.log(`# # # Test 1 PASSED`);
  } catch (error) {
    logError(error, "# # # Test 1 FAILED!");
  }
  console.log("##################### QUEUE ########################");
})();

File: DataStructsAndAlgos/implementations/python/src/Queue.test.py
from ds.Queue.queue import Queue
from Testing.test import COMPARE_VALUES, COMPARE_LISTS


def testOne():
    try:
        queue = Queue()
        queue.push(1)
        queue.push(2)
        queue.push(3)
        COMPARE_VALUES(True, queue.search(1))
        COMPARE_VALUES(True, queue.search(2))
        COMPARE_VALUES(True, queue.search(3))
        COMPARE_VALUES(False, queue.search(4))
        COMPARE_VALUES(False, queue.search('Hi!'))
        COMPARE_VALUES(3, queue.getLength())
        COMPARE_VALUES(1, queue.peek())
        COMPARE_VALUES(1, queue.pop())
        COMPARE_VALUES(2, queue.peek())
        COMPARE_VALUES(2, queue.getLength())
        COMPARE_VALUES(2, queue.pop())
        COMPARE_VALUES(3, queue.peek())
        COMPARE_VALUES(1, queue.getLength())
        COMPARE_VALUES(3, queue.pop())
        COMPARE_VALUES(None, queue.pop())
        COMPARE_VALUES(None, queue.pop())
        COMPARE_VALUES(None, queue.peek())
        COMPARE_VALUES(0, queue.getLength())

        pass
    except Exception as err:
        print(err)
        print('Test 1 FAILED!')
    else:
        print('Test 1 Passed!')
    finally:
        print('Test 1 Complete.')


if __name__ == "__main__":
    print('######################### Queue #########################')
    testOne()
    print('######################### Queue #########################')
